Assets/Scripts/Localization/LocalizationTable.cs

```diff
@@ -6,193 +6,118 @@
-                    }
-                    else
-                    {
-                        // The loop is searching for a value. Add the current char, regardless of what it is.
-                        currentValue += c;
+                        throw new InvalidOperationException(string.Format("Invalid format of localization string. Actual {0}", line));
                     }
+                    localizationTable[localizationCode].Add(keyValuePair[0], keyValuePair[1]);
                 }
-
-                // Add the new key+value to the localization table.
-                localizationTable.Add(localizationCode + "_" + currentKey, currentValue);
             }
-
-            registeredLanguages.Add(localizationCode);
+            catch (FileNotFoundException exception)
+            {
+                Logger.LogException(new Exception(string.Format("There is no localization file for {0}", localizationCode), exception));
+            }
         }
 
-        /**
-         * <summary>
-         * Returns the localization for the given key, or the key itself, if no translation exists.
-         * </summary>
-         * 
-         * <para>
-         * string key: The key that should be searched for.
-         * </para>
-         * 
-         * <para>
-         * params string[] additionalValues: The values that should be inserted.
-         * </para>
-         */
+        /// <summary>
+        /// Returns the localization for the given key, or the key itself, if no translation exists.
+        /// </summary>
+        /// <param name="key">The key that should be searched for.</param>
+        /// <param name="additionalValues">The values that should be inserted.</param>
+        /// <returns></returns>
         public static string GetLocalization(string key, params string[] additionalValues)
         {
             // Return the localization of the advanced method.
-            return GetLocalization(key, FallbackMode.ReturnEnglish, currentLanguage, additionalValues);
+            return GetLocalization(key, FallbackMode.ReturnDefaultLanguage, currentLanguage, additionalValues);
         }
 
-        /**
-         * <summary>
-         * Returns the localization for the given key, or the key itself, if no translation exists.
-         * </summary>
-         */
-        public static string GetLocalization(string key, FallbackMode fallbackMode, string language, params string[] additionalValues)
+        /// <summary>
+        /// Returns the localization for the given key, or the key itself, if no translation exists.
+        /// </summary>
+        private static string GetLocalization(string key, FallbackMode fallbackMode, string language, params string[] additionalValues)
         {
-            key = language + "_" + key;
             string value;
-            if (localizationTable.TryGetValue(key, out value))
+            if (localizationTable.ContainsKey(language) && localizationTable[language].TryGetValue(key, out value))
```

Comment:
This could be slightly misleading. If you select a language that doesnt exist, it will log the _key_ as missing. I think this was discussed in another issue. I'm willing to let this one slip for this pull request, it can be fixed later. :)